client/extensions/woocommerce/app/products/product-create.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import { getCurrentlyEditingProduct } from '../../state/ui/products/selectors';
+import { editProduct, editProductAttribute } from '../../state/ui/products/actions';
+import ProductForm from './product-form';
+
+class ProductCreate extends Component {
+
+	componentDidMount() {
+		const { product } = this.props;
+
+		if ( ! product ) {
+			this.props.editProduct( null, {
+				type: 'simple'
+			} );
+		}
+	}
+
+	componentWillUnmount() {
+		// TODO: Remove the product we added here from the edit state.
+	}
+
+	render() {
+		const { product } = this.props;
+
+		return (
+			<ProductForm
+				product={ product || {} }
```

Comment:
I don't think we can have simply {} as the fallback. If you do a 

```
localStorage.clear();
indexedDB.deleteDatabase( 'calypso' )
```
and then refresh the (http://calypso.localhost:3000/store/products/add) page, you get the following errors:

<img width="964" alt="screen shot 2017-04-21 at 4 45 08 pm" src="https://cloud.githubusercontent.com/assets/1595739/25299294/33996f7a-26b2-11e7-866b-cdea9bdfeb34.png">